### ISSUE TITLE
TCP connection & address mapping

### ISSUE THREAD

**Issue by lacourte** [SamCooper/TCPIP_SPEC_RIDS#1]
We have discussed this issue by email.
The attached document is a proposal solution.
I was not allowed to push the original Word document here (doc), so I push a docx version. You may also refer to the document I sent before by mail.

[Mission Operations - MAL binding to TCPIP Transport and Split Binary Encoding_finalDraft_SADT.docx](https://github.com/SamCooper/TCPIP_SPEC_RIDS/files/476523/Mission.Operations.-.MAL.binding.to.TCPIP.Transport.and.Split.Binary.Encoding_finalDraft_SADT.docx)

**Comment by dariolucia** [comment]
[Mission Operations - MAL binding to TCPIP Transport and Split Binary Encoding_finalDraft_SADT_DL.docx](https://github.com/SamCooper/TCPIP_SPEC_RIDS/files/511221/Mission.Operations.-.MAL.binding.to.TCPIP.Transport.and.Split.Binary.Encoding_finalDraft_SADT_DL.docx)
Following the discussions and the agreements, find attached the revised version of the specification that we are implementing at the moment. It covers what has been agreed and updated by ScalAgent. In addition, I would like to draw your attention to one potential ambiguity that we found in section 3.4.7, i.e. encoding of the list of identifiers ('domain' field). In order to ensure the correct encoding, the 'Presence field' cannot be encoded in the bit field of the body, but a fixed-length encoding must be used. 
Let me know if the suggestion is accepted.

**Comment by lacourte** [comment]
I agree with the problem raised by Dario, and I also agree with the proposed change.

**Comment by SamCooper** [comment]
I'm not sure I completely understand what you are proposing for 3.4.7. Are you suggesting that the presence field for each part of the Identifier list is removed for this one field? Or something else? Could you maybe give an example?
At the moment it is, in my head, encoded as:

|Flags (inc Domain flag)| <other fields> | Domain Length | Part 1 Presence | Part 1 value| Part2 Pre....

**Comment by dariolucia** [comment]
No, I am suggesting that the 'presence field' for the List<Identifier> domain in the header is encoded as a standard 1-byte boolean, instead of a boolean as defined for the Split Binary Encoding. Encoding it as a boolean according to the SBE would mean to have a bit field, which we do not have in the header.

It will be encoded in the following way

`|Flags ... | .... | Domain Length (Unsigned VarInt) | Part 1 Presence (1 byte) |  Part 1 Value | Part 2 Presence (1 byte) | Part 2 Value ...`

For standard lists of nullable elements that are present in the body, the Split Binary Encoding is used, so the presence flag is encoded in the bit field

`| Bit Field Length (UInteger) | Bit Field (mult-octet) | .... | List Length (UInteger) | Part 1 Value | Part 2 Value | ...`

In the latter case, the presence field is encoded inside the bit field.

**Comment by SamCooper** [comment]
Ok, that's kind of what I understood.

**Comment by SamCooper** [comment]
So, just to confirm, the address issue originally raised by ScalAgent will be implemented in both the book and the testbed?

**Comment by dariolucia** [comment]
The address issue led to the proposed change in section 3.3.2 (see document in the 2nd comment). Our TCP/IP implementation has been updated accordingly.
